pyop2/compilation.py

```diff
@@ -54,6 +55,31 @@ def _check_hashes(x, y, datatype):
     return False
 
 
+@collective
+def single_out_a_rank(comm):
+    rank = MPI.Comm.Get_rank(comm)
+    debug('Using file-based mechanism on %d', rank)
+    fname = os.path.join("/tmp", "fb.%d" % rank)
+
+    with open(fname, 'w+') as f:
+        f.write("This is a file.")
+
+    MPI.Comm.Barrier(comm)
+    tbb = []
+    # for dl, sl, fl in os.walk("/tmp"):
+    #     for fi in fl:
+    files = glob.glob("/tmp/fb.[0-9]+")
+    for fi in files:
+        if str(fi).split(".")[0] == "fb":
+            tbb.append(str(fi).split(".")[1])
+
+    os.remove(fname)
```

Comment:
I think you need a barrier before removing the file.  Otherwise rank A may remove its file before rank B reads it.  (It's like shared memory consistency models, only worse, because the hardware gives you even fewer guarantees...).